src/printer.js

```diff
@@ -1124,25 +1136,26 @@ function genericPrintNoParens(path, options, print, args) {
 
       return concat(parts);
     case "SequenceExpression": {
-      const parent = path.getParentNode();
-      const shouldInline =
-        parent.type === "ReturnStatement" ||
-        parent.type === "ForStatement" ||
-        parent.type === "ExpressionStatement";
-
-      if (shouldInline) {
-        return join(", ", path.map(print, "expressions"));
+      const parent = path.getParentNode(0);
+      if (
+        parent.type === "ExpressionStatement" ||
+        parent.type === "ForStatement"
+      ) {
+        // For ExpressionStatements and for-loop heads, which are among
+        // the few places a SequenceExpression appears unparenthesized, we want
+        // to indent expressions after the first.
+        const parts = [];
+        path.each(p => {
+          if (p.getName() === 0) {
```

Comment:
Could do, though this is actually a [documented use](https://github.com/prettier/prettier/blob/a0cadad2220a7843516de5f1ca44f10f0bccc5ad/src/fast-path.js#L93-L94) of getName. Are you sure you want it changed?